nilearn/surface/surface.py

```diff
@@ -1262,33 +1266,51 @@ def __init__(self, left=None, right=None):
                 "Either left or right (or both) must be provided."
             )
 
+        # to ensure that if 1D array are passed as input
+        # they will be saved as 1D array
+        self._squeeze_on_save = {"left": False, "right": False}
```

Comment:
Not too happy with this way of 'tracking' what the object was initialized with just to ensure that 1D array are saved properly to a gifti file, but at least it works...

Well unless, once again, some of the parts are manually set with a 1D array after init.

Reply:
`self._squeeze_on_save` should be a Bool: there should not be a distinction between left and right hemispheres.
And I think it should be configurable at init.